WalletWasabi.Daemon/Rpc/WasabiJsonRpcService.cs

```diff
@@ -188,6 +188,24 @@ public object GetStatus()
 		};
 	}
 
+	[JsonRpcMethod("getcoinjoinstatus")]
+	public string GetCoinjoinStatus()
+	{
+		var activeWallet = Guard.NotNull(nameof(ActiveWallet), ActiveWallet);
+
+		AssertWalletIsLoaded();
+		var coinJoinManager = Global.HostedServices.Get<CoinJoinManager>();
+		_ = coinJoinManager.TryGetWalletStatus(activeWallet.WalletName, out var walletCoinjoinStatus);
```

Comment:
This is an anti-pattern and some devs could copy it. The idea behind the `TryGet` pattern is to do not use the potentially `null` value to control flow but the returned boolean instead. Otherwise it could make much more sense to remove the `Try` part and simply return a nullable whatever, for example, in this case it could be better to have a  `public CoinJoinClientState? GetWalletStatus(string walletName)`  what sucks bad but at list doesn't pretend to implement a pattern to ignore it later. In summary, if you have a TryGetXXXX do not ignore the returned value.

Now, the next question is why do we have a `TryGetWalletStatus`!? I mean, we the `Try` part? An open wallet should have an state. The state can be idle, in progress or whatever other state we have but it can never be null or have no status or status unknown.

Reply:
Thank you for the explanation.